configs/webpack.common.config.js

```diff
@@ -1,36 +1,76 @@
 const path = require('path');
 const os = require('os');
+const webpack = require('webpack');
+const WebpackShellPlugin = require('webpack-synchronizable-shell-plugin');
+const HtmlWebpackPlugin = require('html-webpack-plugin');
 
-const { MODULES_DIR } = require('./constants');
+const logger = require('../server/logger');
+const { MODULES_DIR, SRC_DIR, SERVER_DIR } = require('./constants');
 
 // can be passed by server process via bcfg interface
 // or passed manually when running webpack from command line
 // defaults to `~/.bpanel`
 const bpanelPrefix =
   process.env.BPANEL_PREFIX || path.resolve(os.homedir(), '.bpanel');
 
-module.exports = {
-  resolve: {
-    symlinks: false,
-    extensions: ['-browser.js', '.js', '.json', '.jsx'],
-    // list of aliases are what packages plugins can list as peerDeps
-    // this helps simplify plugin packages and ensures that parent classes
-    // all point to the same instance, e.g bcoin.TX will be same for all plugins
-    alias: {
-      bcoin$: `${MODULES_DIR}/bcoin/lib/bcoin-browser`,
-      bcash$: `${MODULES_DIR}/bcash/lib/bcoin-browser`,
-      hsd$: `${MODULES_DIR}/hsd/lib/hsd-browser`,
-      bledger: `${MODULES_DIR}/bledger/lib/bledger-browser`,
-      bmultisig: `${MODULES_DIR}/bmultisig/lib/bmultisig-browser`,
-      react: `${MODULES_DIR}/react`,
-      'react-router': `${MODULES_DIR}/react-router`,
-      'react-router-dom': `${MODULES_DIR}/react-router-dom`,
-      'react-redux': `${MODULES_DIR}/react-redux`,
-      'react-loadable': `${MODULES_DIR}/react-loadable`,
-      '&local': path.resolve(bpanelPrefix, 'local_plugins'),
-      '@bpanel/bpanel-utils': `${MODULES_DIR}/@bpanel/bpanel-utils`,
-      '@bpanel/bpanel-ui': `${MODULES_DIR}/@bpanel/bpanel-ui`,
-      tinycolor: 'tinycolor2'
-    }
+let MY_SECRETS = {};
+
+module.exports = () => {
```

Comment:
Any reason to turn this into a function if it takes no arguments?

Reply:
So that I could set the secrets constant within the scope of the function rather than as a global at the top of the file.

Reply:
gotcha

Reply:
Agreed! updated.